packages/react-button/src/components/Button/useButtonBehavior.ts

```diff
@@ -0,0 +1,31 @@
+import { buttonBehavior } from '@fluentui/accessibility';
+import { getCode } from '@fluentui/keyboard-key';
+import { ButtonProps, ButtonState } from './Button.types';
+
+export const useButtonBehavior = (props: ButtonProps): Partial<ButtonState> => {
+  const { as, disabled, loading, onClick, onKeyDown } = props;
+
+  const { attributes, keyActions } = buttonBehavior({ as: as as string, disabled, loading });
+
+  let _onKeyDown = onKeyDown;
+
+  const keyCombinations = keyActions?.root?.performClick?.keyCombinations;
+  if (keyCombinations) {
+    _onKeyDown = (ev: React.KeyboardEvent<HTMLButtonElement> & React.MouseEvent<HTMLButtonElement>) => {
+      if (onKeyDown) {
+        onKeyDown(ev);
+      }
+
+      if (onClick) {
+        const eventCode = getCode(ev);
+        for (const keyCombination of keyCombinations) {
+          if (eventCode === keyCombination.keyCode) {
```

Comment:
i feel this logic to check key match will be needed in many places in the future. can we extract? or maybe we already have a helper

Reply:
I extracted it into its own hook in `@uifabric/react-hooks`.